marketpulse/main/templates/fxosprice_all.html

```diff
@@ -69,6 +69,11 @@ <h3 class="title">Contributions</h3>
                  class="btn btn-default btn-xs">Details</a>
               <a href="{{ url('main:clone_contribution', contribution.pk) }}"
                  class="btn btn-default btn-xs">Clone</a>
+              {% if contribution.user != user %}
+                <a href="{{ url('main:upvote_contribution') }}"
+                   class="btn btn-default btn-xs upvote-button"
+                   data-contribution-id="{{ contribution.id }}">Upvote</a>
+              {% endif %}
```

Comment:
I would suggest we leave only `Details` button here and move all actions under the detail view. See #147. In any case, I think it's safer to "force" a user to see the full contribution in order to upvote.